src/Hazelcast.Net/FlakeId/Batch.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading;
+
+namespace Hazelcast.FlakeId
+{
+    internal class Batch
+    {
+        private readonly long _increment;
+        private readonly long _max;
+        private readonly DateTime? _invalidSince;
```

Comment:
Can be a non-`null` `DateTime` and use `DateTime.MaxValue` when the `validityPeriod` is infinite?

Reply:
`DateTime.MaxValue` is not "really" infinite. Any reason to avoid `null`?

Reply:
code simplicity but I'm ok if we keep it like it is